ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -534,7 +534,7 @@ private Map<String, Throttler> getThrottlers(StoreConfig config, Time time) {
    */
   private void checkMountPathAccessible() throws StoreException {
     File mountPath = new File(disk.getMountPath());
-    if (!mountPath.exists()) {
+    if (!mountPath.exists() && diskHealthStatus != DiskHealthStatus.MOUNT_NOT_ACCESSIBLE) {
```

Comment:
I am afraid this is not going to fix the issue. 

So first time the disk checker is running, `diskHealthStatus` is null so it doesn't equal to MOUNT_NOT_ACCESSIBLE, it will throw an exception and change the `diskHealthStatus` to MOUNT_NOT_ACCESSIBLE.

Second time the disk checker is running, `diskHealthStatus` is MOUNT_NOT_ACCESSIABLE, so this method won't throw any exception. However, the createFileChannel method would proceed to create a file and fail. Now the `diskHealthStatus` will be set to `CREATE_EXCEPTION`.

Third time the disk checker is running, `diskHealthStatus` does not equal to MOUNT_NOT_ACCESSIBLE again. And the cycle goes on and on.

I think we need to stop or slow down the frequency of disk checker when we see any exceptions.

Reply:
You're absolutely right. I had that logic in my local branch but hadn't pushed it. In this new commit, checkMountPathAccessible returns whether or not it's accessible. If the function returns False we would then return null in createFileChannel which wouldn't change the state of `diskHealthStatus` and ultimately not return a repeated exception since `diskHealthStatus` is still MOUNT_NOT_ACCESSIBLE. If it's true, we would continue to attempt to create a file channel.